pkg/s3/bucket.go

```diff
@@ -179,44 +183,73 @@ func TagBucket(s3Client *s3.S3, bucketName string, backUpLocation string, infraN
 	return nil
 }
 
-// FindExistingBucket looks for an S3 bucket matching this cluster's velero tags
-// and infrastructure tags. If a matching bucket is found, the bucket name is returned.
-func FindExistingBucket(s3Client *s3.S3, infraName string) (string, error) {
-	// List all buckets associated with this cluster's AWS account.
+// ListBuckets lists all buckets in the AWS account.
+func ListBuckets(s3Client Client) (*s3.ListBucketsOutput, error) {
 	input := &s3.ListBucketsInput{}
 	result, err := s3Client.ListBuckets(input)
 	if err != nil {
 		fmt.Println(err.Error())
-		return "", err
+		return result, err
 	}
+	return result, nil
+}
 
-	for _, bucket := range result.Buckets {
-		request := &s3.GetBucketTaggingInput{
-			Bucket: aws.String(*bucket.Name),
+// ListBucketTags returns a list of s3.GetBucketTagging objects, one for each bucket.
+// If the bucket is not readable, or has no tags, an empty TagSet is returned for that bucket.
+func ListBucketTags(s3Client Client, bucketlist *s3.ListBucketsOutput) (map[string]*s3.GetBucketTaggingOutput, error) {
+	taglist := make(map[string]*s3.GetBucketTaggingOutput)
+	emptyTagSet := &s3.GetBucketTaggingOutput{
+		TagSet: []*s3.Tag{},
+	}
+	for _, bucket := range bucketlist.Buckets {
+		taglist[*bucket.Name] = emptyTagSet
```

Comment:
The other thing is, do we really need to return an empty tag set, as opposed to just `continue`ing?

Reply:
Good catch! I thought I had needed an empty TagSet because of this 

https://github.com/openshift/managed-velero-operator/blob/f04974b60f93eeb3442c353f3d3b231b4eb8b497/pkg/s3/bucket.go#L236

But I tested without it, and it works! It's an improvement because it's more simple and straightforward to read.

Tested in stage with this image `quay.io/sedgar/managed-velero-operator:v0.1.68-89f0787`